node/node.go

```diff
@@ -412,10 +413,31 @@ func (n *Node) initChainManager(avaxAssetID ids.ID) error {
 	criticalChains := ids.Set{}
 	criticalChains.Add(constants.PlatformChainID, createAVMTx.ID())
 
-	n.chainManager, err = chains.New(
+	timeoutManager := timeout.Manager{}
+	if err := timeoutManager.Initialize(
+		n.Config.NetworkInitialTimeout,
+		n.Config.NetworkMinimumTimeout,
+		n.Config.NetworkMaximumTimeout,
+		n.Config.NetworkTimeoutMultiplier,
+		n.Config.NetworkTimeoutReduction,
+		"gecko",
+		n.Config.ConsensusParams.Metrics,
+	); err != nil {
+		return err
+	}
+	go n.Log.RecoverAndPanic(timeoutManager.Dispatch)
+
+	n.Config.ConsensusRouter.Initialize(
+		n.Log,
+		&timeoutManager,
+		n.Config.ConsensusGossipFrequency,
+		n.Config.ConsensusShutdownTimeout,
+	)
+
+	n.chainManager = chains.New(
```

Comment:
We should probably just pass an arg struct rather than passing args like this

Reply:
Agreed.